src/main.rs

```diff
@@ -897,10 +910,16 @@ fn run() -> Result<(), failure::Error> {
         let manifest = settings::toml::Manifest::new(config_path)?;
         let env = matches.value_of("env");
         let target = manifest.get_target(env)?;
-
         let user = settings::global_user::GlobalUser::new()?;
 
-        commands::tail::start(&target, &user)?;
+        let tunnel_port: Option<u16> = matches
+            .value_of("tunnel_port")
+            .map(|p| p.parse().expect("--port expects a number"));
+        let metrics_port: Option<u16> = matches
+            .value_of("metrics_port")
+            .map(|p| p.parse().expect("--metrics expects a number"));
```

Comment:
can't wait until clap 3 comes out and we can make serde do all of this stuff up front

Reply:
agreed.